client/chat/channel-info-selector.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback } from 'react'
+import {
+  BasicChannelInfo,
+  DetailedChannelInfo,
+  JoinedChannelInfo,
+  SbChannelId,
+} from '../../common/chat'
+import { RootState } from '../root-reducer'
+
+/**
+ * Combined (and flattened) channel info that simplifies the usage in React components.
+ */
+export type ClientChannelInfo = BasicChannelInfo &
+  Partial<DetailedChannelInfo> &
+  Partial<JoinedChannelInfo>
+
+/**
+ * Returns a function for use with `useAppSelector` that finds all the available information about
```

Comment:
I think this is a bad pattern, and doesn't seem particularly easier to use. Please just use the parts that you actually need. By constructing it this way you will force the component to always re-render.

Even if you memoized it, you would still force rerenders of a whole component even if they only depend on one small piece of this state. Just select the pieces separately.